src/oss-find-source/FindSourceTool.cs

```diff
@@ -6,91 +6,77 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using CommandLine;
+using CommandLine.Text;
 using Microsoft.CodeAnalysis.Sarif;
 using Microsoft.CST.OpenSource.Shared;
+using static Microsoft.CST.OpenSource.Shared.OutputBuilder;
 
 namespace Microsoft.CST.OpenSource
 {
     public class FindSourceTool : OSSGadget
     {
-        /// <summary>
-        /// Name of this tool.
-        /// </summary>
-        private const string TOOL_NAME = "oss-find-source";
-
-        /// <summary>
-        /// Holds the version string, from the assembly.
-        /// </summary>
-        private static readonly string VERSION = typeof(FindSourceTool).Assembly?.GetName().Version?.ToString() ?? string.Empty;
-
-        /// <summary>
-        /// Command line options
-        /// </summary>
-        private readonly Dictionary<string, object?> Options = new Dictionary<string, object?>()
+        public class Options
         {
-            { "target", new List<string>() },
-            { "show-all", false },
-            { "format", "text" },
-            { "output-file", null }
-        };
-
-        static void Main(string[] args)
-        {
-            var findSourceTool = new FindSourceTool();
-            Logger.Debug($"Microsoft OSS Gadget - {TOOL_NAME} {VERSION}");
-            findSourceTool.ParseOptions(args);
-
-            // output to console or file?
-            bool redirectConsole = !string.IsNullOrEmpty((string?)findSourceTool.Options["output-file"]);
-            if(redirectConsole && findSourceTool.Options["output-file"] is string outputLoc)
+#pragma warning disable CS8618 // Non-nullable field is uninitialized. Consider declaring as nullable - commandlineparser doesnt handle nullable fields
```

Comment:
Please address these nullable issues instead of disabling the help that detects them.

Reply:
The problem when I make the class properties nullable is that the CommandLineParser throws InvalidCastException - I guess not all libraries handle nullables because it is a newer feature? Do you know any way around this?

E:\root\OSSGadget\src\oss-find-source\bin\Debug\netcoreapp3.1>.\oss-find-source.exe
Unhandled exception. System.InvalidCastException: Unable to cast object of type 'System.Runtime.CompilerServices.NullableAttribute' to type 'CommandLine.Text.UsageAttribute'.
   at CommandLine.Core.ReflectionExtensions.<>c.<GetUsageData>b__2_3(<>f__AnonymousType1`2 <>h__TransparentIdentifier0)
   at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   at System.Linq.Enumerable.SingleOrDefault[TSource](IEnumerable`1 source)
   at CommandLine.Core.ReflectionExtensions.GetUsageData(Type type)
   at CommandLine.Text.HelpText.GetUsageFromType(Type type)
   at CommandLine.Text.HelpText.RenderUsageTextAsLines[T](ParserResult`1 parserResult, Func`2 mapperFunc)+MoveNext()
   at CSharpx.EnumerableExtensions.ToMaybe[T](IEnumerable`1 source)
   at CommandLine.Text.HelpText.AutoBuild[T](ParserResult`1 parserResult, Func`2 onError, Func`2 onExample, Boolean verbsIndex, Int32 maxDisplayWidth)
   at CommandLine.Text.HelpText.AutoBuild[T](ParserResult`1 parserResult, Func`2 onError, Int32 maxDisplayWidth)
   at Microsoft.CST.OpenSource.OSSGadget.DisplayHelp[T](ParserResult`1 result, IEnumerable`1 errs) in E:\root\OSSGadget\src\Shared\OSSGadget.cs:line 34
   at Microsoft.CST.OpenSource.OSSGadget.<>c__DisplayClass8_0`1.<ParseOptions>b__0(IEnumerable`1 errs) in E:\root\OSSGadget\src\Shared\OSSGadget.cs:line 56
   at CommandLine.ParserResultExtensions.WithNotParsed[T](ParserResult`1 result, Action`1 action)
   at Microsoft.CST.OpenSource.OSSGadget.ParseOptions[T](String[] args) in E:\root\OSSGadget\src\Shared\OSSGadget.cs:line 56
   at Microsoft.CST.OpenSource.FindSourceTool.Main(String[] args) in E:\root\OSSGadget\src\oss-find-source\FindSourceTool.cs:line 53
   at Microsoft.CST.OpenSource.FindSourceTool.<Main>(String[] args)

Reply:
I added the nullable annotations and cannot reproduce this.  I'll push my fixed version.

If you do *need* to disable nullability somewhere, use 

```
#nullable disable
<code>
#nullable enable
```

Rather than disabling the warnings.  

The disabling of the warning you did also shouldn't have actually changed if this worked or not, that just change the squiggled that show up in the IDE.

Reply:
I dont see this issue now in oss-find-source but it happens in oss-characteristic

Reply:
So this is weirdly only happening on the OutputFile parameter.  I took a look and you check `string.IsNullOrEmpty()` so we can actually declare this as non-nullable and set the default value to the empty string.  I've done that and pushed a change.

Reply:
Resolving this since I have made the required changes.